src/Chapter2.hs

```diff
@@ -336,7 +348,7 @@ from it!
 ghci> :l src/Chapter2.hs
 -}
 subList :: Int -> Int -> [a] -> [a]
-subList = error "subList: Not implemented!"
+subList x y z = if x < 0 || y < 0 then [] else take ( y - x + 1) ( drop x z)
```

Comment:
Your implementation is correct when the right bound is lower than the left. But I think you can implement it more efficiently by adding a separate case to your check :slightly_smiling_face:

You can also use guards here instead of `if-then-else` expressions for cleaner and more concise code 💂🏻

Reply:
Thanks. Yes, I see with patten matching this is simpler.